inbm/dispatcher-agent/dispatcher/sota/sota.py

```diff
@@ -356,7 +356,7 @@ def execute_from_manifest(self,
                     self._update_logger.status = OTA_PENDING
                     self._update_logger.error = ""
                 self._update_logger.save_log()
-                if self.sota_mode == 'download-only' or self._device_reboot in ["No", "N", "n", "no", "NO"]:  # pragma: no cover
+                if self.sota_mode == 'download-only' or not self._is_reboot_device():
                     self._dispatcher_broker.telemetry("No reboot (SOTA pass)")
                 else:
                     self._dispatcher_broker.telemetry("Going to reboot (SOTA pass)")
```

Comment:
**Suggestion:** The execute_from_manifest method contains a condition that checks if self.sota_mode is 'download-only' or if the device should not reboot. This condition is used to determine whether to send a telemetry message or to initiate a reboot. However, the condition is not grouped, which could lead to unexpected behavior if the order of evaluation is not as intended. It is a good practice to group conditions to ensure the correct order of evaluation and improve code readability. [maintainability]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
                if (self.sota_mode == 'download-only') or (not self._is_reboot_device()):
                    self._dispatcher_broker.telemetry("No reboot (SOTA pass)")
                else:
                    self._dispatcher_broker.telemetry("Going to reboot (SOTA pass)")
```